--- a/configure.ac
+++ b/configure.ac
@@ -338,11 +338,11 @@
 GCC_VISIBILITY_CC
 
 # Host-dependant definitions
-DEFAULT_DUFLAGS=" -sk --apparent-size"
-DEFAULT_SEDINPLACEFLAGS=" --follow-symlinks -i"
+DEFAULT_DUFLAGS=" -sk"
+DEFAULT_SEDINPLACEFLAGS=" -i"
 INODECMD="stat -c '%i %n'"
 OWNERCMD="stat -c '%u:%g'"
 MODECMD="stat -c '%a'"
 STRIP_BINARIES="--strip-all"
 STRIP_SHARED="--strip-unneeded"
 STRIP_STATIC="--strip-debug"
